src/auditwheel/repair.py

```diff
@@ -126,6 +133,18 @@ def repair_wheel(
             extensions = external_refs_by_fn.keys()
             strip_symbols(itertools.chain(libs_to_strip, extensions))
 
+        # If we grafted packages with identities we add an SBOM to the wheel.
+        # We recalculate the checksum at this point because there can be
+        # modifications to libraries during patching.
+        sbom_data = create_sbom_for_wheel(
+            wheel_fname=wheel_fname,
```

Comment:
this should likely take the updated filename in order to restrict the sbom component purl to the current wheel:
```suggestion
            wheel_fname=output_wheel.name,
```